src/components/file-viewer/file-viewer.tsx

```diff
@@ -0,0 +1,390 @@
+import { FileType, OfficeViewer } from './file-viewer.types';
+
+/**
+ * This is a smart component that automatically detects
+ * the most common file types such as image, audio, video, and text,
+ * and properly displays them in the browser.
+ * The component is also capable to render the most common office files.
+ *
+ * :::note
+ * Image files will always be contained in their containers, which means
+ * they automatically increases or decreases in size to fill their containing box
+ * whilst preserving their aspect-ratio.
+ *
+ * Text and PDF files will also always respect the width and height of the
+ * container in which the `limel-file-viewer` is loaded.
+ * :::
+ *
+ * For some file types such as text and images, the component will display a
+ * download button and a button to open the file in a new browser tab.
+ * This will allow users to preview the file in a fullscreen more with the
+ * browser and take advantage of for example native zooming and panning
+ * functionalities.
+ *
+ *:::important
+ * The download button will not work here in this documentation,
+ * due to the security policies of the web browsers,
+ * since example files are not hosted in the same domain.
+ *:::
+ * @exampleComponent limel-example-file-viewer
+ * @exampleComponent limel-example-file-viewer-office
+ * @exampleComponent limel-example-file-viewer-filename
+ * @exampleComponent limel-example-file-viewer-actions
+ * @exampleComponent limel-example-file-viewer-with-picker
+ */
+
+@Component({
+    tag: 'limel-file-viewer',
+    shadow: true,
+    styleUrl: 'file-viewer.scss',
+})
+export class FileViewer {
+    /**
+     * Link to the file
+     */
+    @Prop({ reflect: true })
+    public url: string;
+
+    /**
+     * The name of the file that also contains its extension.
+     * Overrides the filename provided by the URL.
+     * Useful when the URL does not contain the filename.
+     */
+    @Prop({ reflect: true })
+    public filename: string;
+
+    /**
+     * Alternative text for assistive technologies and screen readers,
+     * used for image files.
+     */
+    @Prop({ reflect: true })
+    public alt: string;
+
+    /**
+     * Defines the localization for translations.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * Defines the third-party viewer that should be used to render
+     * the content of office files, such as word processing documents,
+     * presentations, or spreadsheets.
+     */
+    @Prop()
+    public officeViewer: OfficeViewer = 'microsoft-office';
+
+    /**
+     * An array of custom actions that can be displayed
+     * as an action menu on the file which is being displayed.
+     */
+    @Prop()
+    public actions: ListItem[];
+
+    /**
+     *
+     */
+    @Prop()
+    public webserverDomain: string;
+
+    /**
+     * determines if the document is hosted on the same domain as the current webserver
+     */
+    @State()
+    public sameDomain: boolean = false;
```

Comment:
Hi @anderssonjohan . Do you remember why we added these?

Reply:
The item in the PR checklist is related to this security feature of the anchor download attribute:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attributes

> Note:
>
> `download` only works for [same-origin URLs](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy), or the blob: and data: schemes.

That means we render a regular anchor element conditionally (in `renderSameDomainDownload`) when we can't offer automatic download via javascript.
Since there's no commit history from before I don't recall where this comes from but the changes in this PR seems to do the opposite? 🤔 If not same domain, then don't use `<a download=...`, but the button without `download` seems to be rendered when _it is_ the same domain. 🤯

Reply:
What you would need to test is downloading a file from some other website when using the component on a website that is not localhost. I.e., try download a file from another website when running the component in the docs site at github.io.

Reply:
I see. Thanks. Currently the `download` attribute is removed from the link, both in the local build, and in the docs site at github.io. It doesn't help that I add it manually in the Dom neither. So it's a bit hard to test what you said. I think we might just skip that for now to let the PR get merged, and later add the download as a feature @Befkadu1 . Unless you have a lot of energy to troubleshoot this 😅

Reply:
> Currently the download attribute is removed from the link

UPDATE: This is because the code says: `download={this.filename}` and there is often no file name.

Reply:
Sure. When it comes to documents in systems like Lime CRM, the download will happen from other sites than the site running the web app, hence why we added this (I guess).

Reply:
I think the main intention for creating this component was not to force users to download files to see them. That was the main "pain" so to speak. In CRM we can always make our own Download button as a custom action and show them in those cases. Also if the user is using LiveDocs and SharePoint, we most probably will use their <iframe/> embedding code instead, and implement the download feature if their viewer already doesn't have it. I think they will have it.

I'd say the main purpose of this component isn't really downloading. It's viewing the file's content.